test/es6/destructuring_catch.js

```diff
@@ -189,6 +189,93 @@ var tests = [
             }
         })();
      }
+   },
+   {
+        name: "Function definitions in catch's parameter",
+        body: function () {
+            (function() {
+                try {
+                    var c = 10;
+                    throw ['inside'];
+                } catch ([x, y = function() { return c; }]) {
+                    assert.areEqual(y(), 10, "Function should be able to capture symbols from try's body properly");
+                    assert.areEqual(x, 'inside', "Function should be able to capture symbols from try's body properly");
+                } 
+            })();
+
+            (function() {
+                try {
+                    throw [];
+                } catch ([x = 10, y = function() { return x; }]) {
+                    assert.areEqual(y(), 10, "Function should be able to capture symbols from catch's param");
+                } 
+            })();
+
+            (function() {
+                try {
+                    throw [];
+                } catch ([x = 10, y = function() { return x; }]) {
+                    eval("");
+                    assert.areEqual(y(), 10, "Function should be able to capture symbols from catch's param");
+                } 
+            })();
+
+            (function() {
+                try {
+                    throw {};
+                } catch ({x = 10, y = function() { return x; }}) {
+                    assert.areEqual(y(), 10, "Function should be able to capture symbols from catch's param");
+                } 
+            })();
+
+            (function() {
+                try {
+                    throw ['inside', {}];
+                } catch ([x = 10, { y = function() { return x; } }]) {
+                    assert.areEqual(y(), 'inside', "Function should be able to capture symbols from catch's param");
+                } 
+            })();
+
+            (function() {
+                try {
+                    throw ['inside', {}];
+                } catch ([x, { y = () => arguments[0] }]) {
+                    assert.areEqual(y(), 10, "Function should be able to capture the arguments symbol from the parent function");
+                    assert.areEqual(x, 'inside', "Function should be able to capture symbols from try's body properly");
+                } 
+            })(10);
+
+            (function(a = 1, b = () => a) {

```

Comment:
good to repeat few tests with 'eval' in it (in the try block, catch block and eval in the function on param pattern).